xarray/core/dataset.py

```diff
@@ -1478,7 +1478,7 @@ def __iter__(self) -> Iterator[Hashable]:
     if TYPE_CHECKING:
         # needed because __getattr__ is returning Any and otherwise
         # this class counts as part of the SupportsArray Protocol
-        __array__ = None
+        __array__ = None  # type: ignore[var-annotated,unused-ignore]
```

Comment:
FYI I was getting this locally but not in CI — not sure why — but on balance seems reasonable to exclude. Possible a dependency issue; we can try removing some of these `ignore[unused-ignore]` in the future manually, given they definitionally elude our other checks...